### ISSUE TITLE
Unexpected error for data collection in a sub-directory

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS/validate#240]
**Describe the bug**
Validate throws an error for `unallowed directory name` even though it is the base data collection.

**To Reproduce**
1. download bundle from `pds-gamma:/data/local/starbase/data/pds4/test-bundles/pds4-compil-comet-v1.0/`
2. run validate with the pds4.bundle rule enabled

**Expected behavior**
This should validate successfully because it is a data collection. This is a misinterpretation of the PDS4 Standard Section 2B.2.2 and 6C.2.3, which calls this out as an example, not a requirement. We should just throw a WARNING message here:

```
WARNING  [warn.sub_directory.unallowed_name]   Directory name is reserved and not in the recommended location in the bundle root directory.
```

Applicable requirements:
* 🦄 #239

**Comment by qchaupds** [comment]
As Jordan had pointed out, currently if an unexpected sub directory is found outside of the root directory, it thinks it is an error:
and report UNALLOWED_BUNDLE_SUBDIRECTORY_NAME in src/main/java/gov/nasa/pds/tools/validate/rule/pds4/PDS4Problems.java

`public static final ProblemDefinition UNALLOWED_BUNDLE_SUBDIRECTORY_NAME = new ProblemDefinition(
            ExceptionType.ERROR,
            ProblemType.UNALLOWED_BUNDLE_SUBDIR_NAME,
            "Directory name is not allowed except in the bundle root directory",
            Standard.PDS4_STANDARDS_REFERENCE,
            "6C.2.2"
    );`

So a preferred ExceptionType is a WARNING with a different message:

`public static final ProblemDefinition UNALLOWED_BUNDLE_SUBDIRECTORY_NAME = new ProblemDefinition(
            ExceptionType.WARNING,
            ProblemType.UNALLOWED_BUNDLE_SUBDIR_NAME,
            "Directory name is reserved and not in the recommended location in the bundle root directory",
            Standard.PDS4_STANDARDS_REFERENCE,
            "6C.2.2"
    );`
